### PR TITLE
Fixed course save button not redirecting

### PR DESCRIPTION
**Pull request by iivic**
@mjfrey  @marjev  @vkaracic 
Easy fix. The problem was detecting the issue.
I had to remove the store.reset() from form_view since the coupon model didn't work correctly. I moved that line in the coupon_model save function so the DuplicateId error wouldn't appear when creating a new coupon. 
I didn't encounter DuplicateId error when creating a new course so this fix should be sufficient. However further testing is needed in both cases to be sure.
Please review and test creating and editing courses and coupons I did but just in case :)

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @iivic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=912

File: ecommerce/static/js/models/course_model.js [hunk 1]
```diff
@@ -301,7 +301,7 @@ define([
              * to external systems, in an atomic fashion. This is desirable to
              * avoid synchronization issues across systems.
              */
-            save: function (options) {
+            save: function (attributes, options) {
```
Comment (vkaracic) [inline ecommerce/static/js/models/course_model.js:304]: Why this change?

Comment (iivic) [reply ecommerce/static/js/models/course_model.js:304]: Because the save call is sending two parameters: attributes that should be changed and options ( success, error, etc. ). Without this change the course save button doesn't call any callback function which should be called.

Comment (vkaracic) [reply ecommerce/static/js/models/course_model.js:304]: So this is the reason why there was no redirect after course save?

Comment (iivic) [reply ecommerce/static/js/models/course_model.js:304]: Exactly.

**Comment by vkaracic** [comment]
:+1:

**Comment by marjev** [comment]
👍
